Gems/Atom/RHI/Vulkan/Code/Source/RHI/ShaderResourceGroupPool.cpp

```diff
@@ -55,7 +55,11 @@ namespace AZ
 
             m_descriptorSetAllocator = RHI::Ptr<DescriptorSetAllocator>(aznew DescriptorSetAllocator);
             // [GFX_TODO] ATOM-16891 - Refactor Descriptor management system
-            const uint32_t descriptorSetsPerPool = 20;
+            uint32_t descriptorSetsPerPool = 20;
+            if (layout.GetGroupSizeForBufferUnboundedArrays() > 0 || layout.GetGroupSizeForImageUnboundedArrays() > 0)
+            {
+                descriptorSetsPerPool = 5;
```

Comment:
Can you add following comment here

"Reducing the initial allowed descriptor sets (i.e SRGs) with unbounded arrays per pool to 5 instead of 20. This significantly helps reduce descriptor waste as we allocate 900k descriptors per an unbounded array entry."

Also make sure to run the full test suite in ASV to test this change. Thanks

Reply:
How do I run that test?

Reply:
Launch AtomSampleViewer with vulkan rhi (-rhi=vulkan)
Automation -> Run Script-> Pick Scripts/_fulltestsuite_.bv.luac -> Run
It will cycle through all the ASV samples taking screenshot and do screenshot comparison. In the end you will get a summary screen where all passes "should" pass. Sometimes you will see failures but they may be false negatives. One way to confirm that a test fail is not due to your changes is if you run the test suite without your changes and they still happen. Once you are happy with the summary screen you can attach it here as confirmation.

Reply:
> Launch AtomSampleViewer with vulkan rhi (-rhi=vulkan) Automation -> Run Script-> Pick Scripts/_fulltestsuite_.bv.luac -> Run It will cycle through all the ASV samples taking screenshot and do screenshot comparison. In the end you will get a summary screen where all passes "should" pass. Sometimes you will see failures but they may be false negatives. One way to confirm that a test fail is not due to your changes is if you run the test suite without your changes and they still happen. Once you are happy with the summary screen you can attach it here as confirmation.

I was trying to open the project but it seems to segfault.
```
Process 21809 stopped
* thread #1, name = 'Editor', stop reason = signal SIGSEGV: invalid address (fault address: 0x1d0)
    frame #0: 0x00007fff375d9c2d libAtom_Feature_Common.Editor.so`AZStd::vector<AZ::RHI::ReflectionNamePair<AZ::RHI::Handle<unsigned int, AZ::RHI::ShaderInputBufferDescriptor> >, AZStd::allocator>::begin(this=0x00000000000001d0) const at vector.h:487:93
   484 	        AZ_FORCE_INLINE size_type               capacity() const    { return m_end - m_start; }
   485 	
   486 	        AZ_FORCE_INLINE iterator                begin()             { return iterator(AZSTD_POINTER_ITERATOR_PARAMS(m_start)); }
-> 487 	        AZ_FORCE_INLINE const_iterator          begin() const       { return const_iterator(AZSTD_POINTER_ITERATOR_PARAMS(m_start)); }
   488 	        AZ_FORCE_INLINE iterator                end()               { return iterator(AZSTD_POINTER_ITERATOR_PARAMS(m_last)); }
   489 	        AZ_FORCE_INLINE const_iterator          end() const         { return const_iterator(AZSTD_POINTER_ITERATOR_PARAMS(m_last)); }
   490 	
(lldb)
```

Reply:
I should be able to test this for you next week as I am trying to wrap up a deliverable this week.

Reply:
That's great to hear @moudgils, thanks for checking! I'll let you do the honors of pressing the 'Squash and merge' button 🙂